### ISSUE TITLE
[no-unused-components] Allow registering components only used in script

### ISSUE THREAD

**Issue by steersbob** [vuejs/eslint-plugin-vue#1478]
**What rule do you want to change?**
vue/no-unused-components

**Does this change cause the rule to produce more or fewer warnings?**
Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
Probably default behavior, but either approach would work.

**Please provide some example code that this change will affect:**

```vue
<template>
<div @click="showDialog">
  click me
</div>
</template>

<script>
import { createDialog } from './dialog-utils';
import DialogComponent from './DialogComponent.vue';

export default {
  components: {
    DialogComponent,
  },
  methods: {
    showDialog() {
      createDialog({ component: DialogComponent })
    },
  },
};
</script>
```

**What does the rule currently do for this code?**
An error is shown for the `DialogComponent` registration, as it is not used in the template.

**What will the rule do after it's changed?**
Consider usage in methods, computed, data, and setup sections as valid.

**Additional context**
The use case is relatively generic: pass a component to a function that presumably uses it in a render function somewhere down the line. Dialogs are used as an example as I'm encountering the false positive when using [quasar custom dialogs](https://quasar.dev/quasar-plugins/dialog).

**Comment by LinusBorg** [comment]
In the described "other usecases", you don't need to register the component, you can remove it as it doesn't do anything.

**Comment by steersbob** [comment]
After some further testing, it looks like you're correct, and I misinterpreted an error I got. Thanks for the clarification!
